File: python3/001_TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}
        for i in range(0, len(nums)):
            v = target - nums[i]
            if v in dic:
                return [i, dic[v]] if i<dic[v] else [dic[v], i]
            dic[nums[i]] = i
            

File: cpp/001_TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> map;
        vector<int> indices;
        for(int i=0; i<nums.size(); i++){
            map.insert(pair<int,int>(nums.at(i),i));
        }
        for(int i=0; i<nums.size(); i++){
            int c = target-nums.at(i);
            if(map.find(c)!=map.end() && map.at(c)!=i){
                indices.push_back(i);
                indices.push_back(map.at(c));
                break;
            }
        }
        
        return indices;
    }
};